posthog/session_recordings/session_recording_helpers.py

```diff
@@ -121,7 +121,7 @@ def preprocess_replay_events(
 
     size_with_headroom = max_size_bytes * 0.95  # Leave 5% headroom
 
-    distinct_id = events[0]["properties"]["distinct_id"]
+    distinct_id = events[0]["distinct_id"]
```

Comment:
```suggestion
    distinct_id = events[0]["properties"].get("distinct_id") or events[0]["distinct_id"]
```

So it's back-compatible at least?

Reply:
Not sure if `or` works here, don't use Python that often =D

Reply:
you can `distinct_id = events[0]["properties"].get("distinct_id", events[0]["distinct_id"])`

but the events have all along had the distinct id at the top level so we don't need backwards compatibility here

Reply:
I know that old events from the old Android SDK don't send as top-level property, if at this point the event is already normalized (with back-filling high-level property), then all good.

Reply:
> I know that old events from the old Android SDK don't send as top-level property,
🤯 

I wouldn't have thought that would get through the pipeline... OK, I'm going to write tests for all the possible combinations explicitly!